storage/statedb/hasher.go

```diff
@@ -74,48 +74,14 @@ func returnHasherToPool(h *hasher) {
 	hasherPool.Put(h)
 }
 
-// hash collapses a node down into a hash node, also returning a copy of the
-// original node initialized with the computed hash to replace the original one.
-func (h *hasher) hash(n node, db *Database, force bool) (node, node) {
-	// If we're not storing the node, just hashing, use available cached data
-	if hash, dirty := n.cache(); hash != nil {
-		if db == nil {
-			return hash, n
-		}
-		if !dirty {
-			switch n.(type) {
-			case *fullNode, *shortNode:
-				return hash, hash
-			default:
-				return hash, n
-			}
-		}
-	}
-	// Trie not processed yet or needs storage, walk the children
-	collapsed, cached := h.hashChildren(n, db)
-	hashed, lenEncoded := h.store(collapsed, db, force)
-	// Cache the hash of the node for later reuse and remove
-	// the dirty flag in commit mode. It's fine to assign these values directly
-	// without copying the node first because hashChildren copies it.
-	cachedHash, _ := hashed.(hashNode)
-	switch cn := cached.(type) {
-	case *shortNode:
-		cn.flags.hash = cachedHash
-		cn.flags.lenEncoded = lenEncoded
-		if db != nil {
-			cn.flags.dirty = false
-		}
-	case *fullNode:
-		cn.flags.hash = cachedHash
-		cn.flags.lenEncoded = lenEncoded
-		if db != nil {
-			cn.flags.dirty = false
-		}
-	}
-	return hashed, cached
+// hashRoot is similar to hash() but adds special treatment for the root node
+func (h *hasher) hashRoot(n node, db *Database, force bool) (node, node) {
+	return h.hash(n, db, force, true)
 }
 
-func (h *hasher) hashRoot(n node, db *Database, force bool) (node, node) {
+// hash collapses a node down into a hash node, also returning a copy of the
+// original node initialized with the computed hash to replace the original one.
+func (h *hasher) hash(n node, db *Database, force bool, onRoot bool) (node, node) {
```

Comment:
how about this line change to

func (h *hasher) hash(n node, db *Database) (node, node) {
    return hashProc(, db, force, false)
}

func  (h *hasher) hashProc(n node, db *Database, force bool) (node, node) {

Reply:
Okay. I will call the internal function, say `hashNode`.
PTAL when I push another commit.